service/node.go

```
@@ -1762,13 +1762,18 @@ func (s *service) addNewNodeToArray(ctx context.Context, array *StorageArrayConf
 		}
 	}
 
-	host, err := hostAPI.CreateHost(ctx, s.opts.LongNodeName, tenantID)
+	var hostContent types.HostContent
+	if tenantName != "" && tenantID == "" {
+		return status.Error(codes.Internal, utils.GetMessageWithRunID(rid, "Please enter Valid tenant Name : %s", tenantName))
 
-	if err != nil {
-		return err
+	}else{
+		host, err := hostAPI.CreateHost(ctx, s.opts.LongNodeName, tenantID)
+		if err != nil {
+			return err
```

@karthikk92 can we return the error  as in if loop with message

---

nope since if first time adding host entry fails, it will just retry by throwing just error and then finally if not added multiple tries then finally it will throw proper error message when array not added .